project/src/test/resources/docker/sql/create-schema.sql

```diff
@@ -469,3 +475,223 @@ CREATE TABLE dbo.policies(
+      WHERE name = courseName OR acronym = courseAcr
+    );
+
+  SELECT id INTO cid FROM dbo.Course WHERE name = courseName OR acronym = courseAcr;
+
+  -- Creating the Class
+  PERFORM
+    dbo.f_classCalendarCreate (calTerm, cid)
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.Class
+      WHERE courseId = cid AND calendarTerm = calTerm
+    );
+    
+  SELECT id INTO clid FROM dbo.Class WHERE courseId = cid AND calendarTerm = calTerm;
+
+  -- Creating the Class
+  PERFORM
+    dbo.f_classSectionCalendarCreate (clid, calendarSection)
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.ClassSection C
+      WHERE C.classid = clid AND C.id = calendarSection
+    );
+    
+  -- Store ClassSection's ID and Calendar
+  SELECT
+    calendar INTO cscal
+  FROM
+    dbo.ClassSection C
+  WHERE
+    C.classid = clid AND C.id = calendarSection;
+  
+  -- Deletes all of the Class Section's events related to this operation
+  DELETE FROM
+    dbo.CalendarComponent
+  WHERE id IN (
+    SELECT
+      ccs.comp_id
+    FROM
+      dbo.category c JOIN
+      dbo.categories cs ON c.id=cs.value JOIN
+      dbo.calendarcomponents ccs ON ccs.comp_id=cs.comp_id
+    WHERE
+      c.id = categoryid AND ccs.calendar_id = cscal
+  ) AND type = 'E';
+
+  RAISE NOTICE '% course, % class, % section, % category, % language, % calendar', cid, clid, calendarSection, categoryid, langid, cscal;
+END
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE PROCEDURE dbo.sp_createClassSectionEvent(
+  courseName VARCHAR(100),
+  courseAcr VARCHAR(50),
+  calendarSection VARCHAR(50),
+  calTerm VARCHAR(50),
+	summary VARCHAR(50),
+	description VARCHAR(200),
+  lang VARCHAR(10),
+	categ VARCHAR(100),
+	dtstart TIMESTAMP,
+  dtend TIMESTAMP,
+  week_days VARCHAR(20),
+  location VARCHAR(128)
+) AS $$
+#print_strict_params ON
+DECLARE
+  calid INT;
+  component_id INT;
+  langid INT;
+  categoryid INT;
+  dttype INT;
+BEGIN
+
+  -- get target calendar ID (the class section's calendar)
+  SELECT
+    CS.calendar INTO calid
+  FROM
+    dbo.Course C JOIN
+    dbo.Class CL ON C.id = CL.courseid JOIN
+    dbo.ClassSection CS ON CS.classid = CL.id
+  WHERE
+    ( C.acronym = courseAcr OR C.name = courseName )
+    AND CL.calendarTerm = calTerm AND CS.id = calendarSection;
+
+  SELECT L.id INTO langid FROM dbo.Language L WHERE L.name = lang; 
+  SELECT
+    c.id INTO categoryid
+  FROM
+    dbo.category c JOIN
+    dbo.categorylanguage cl ON c.id = cl.category
+  WHERE
+    name = categ AND language = langid;
+
+  -- get the DATE TIME icalendar type
+  SELECT id INTO dttype FROM dbo.icalendardatatype WHERE name = 'DATE-TIME';
+
+  raise notice '% cal, % category, % lang', calid, categoryid, langid;
+	
+  CALL dbo.newEvent(calid,
+      ARRAY[summary],
+      ARRAY[langid],
+      ARRAY[description],
+      ARRAY[langid],
+      ARRAY[categoryid],
+      dtstart,
+      dtend,
+      dttype, -- dtstart dtend type
+      location,
+      week_days, -- by day
+      NULL -- until
```

Comment:
This is the missing `end_date` of the `CalendarTerm` right?

Reply:
Yes!

Reply:
In the previous meeting, a decision was made to implement a dependency between and recurrent event's starting and end dates with the Calendar Term.